src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheCrusader/boss_northrend_beasts.cpp

```diff
@@ -500,10 +502,49 @@ struct npc_snobold_vassal : public ScriptedAI
         }
     }
 
+    Player* SelectNewVictim()
+    {
+        const Map::PlayerList& players = me->GetMap()->GetPlayers();
+        std::vector<Player*> availablePlayers;
+        for (auto i = players.begin(); i != players.end(); ++i)
+        {
+            if (!i->GetSource()->IsAlive())
+            {
+                continue;
+            }
+
+            availablePlayers.push_back(i->GetSource());
+        }
+
+        if (availablePlayers.size() == 0)
+        {
+            return nullptr;
+        }
+
+        srand(me->GetMap()->GetInstanceId());
+        int playerIndex = rand() % availablePlayers.size();
```

Comment:
we have `irand()` which you can use for such purposes. But since we are dealing with a container here we have an even better solution: `Trinity::Containers::SelectRandomContainerElement(availablePlayers);`

you can drop the `cstdlib` include in the process